### ISSUE TITLE
HttpClient creates Quic threads even if HTTP/3 is not in use

### ISSUE THREAD

**Issue by wfurt** [dotnet/runtime#74629]
1. Create simple console app with HTTP and run simple http1.1 request  on Windows 11 or Server 22. 
2. look at application threads.

![image](https://user-images.githubusercontent.com/14356188/186841698-87e4f329-335e-4fa2-971f-28d8f92a7bcd.png)

Application crates 2x threads per core even if HTTP/3 and Quic are not in used. This was originally reported on legacy system with 54 cores. That will create 100+ threads ... for nothing. 

We should figure out how to make `Quic.IsSupported` cheap and defer thread creation until needed. 
This will impact legacy app that has no clue tor intention to use Quic or HTTP/3.

**Comment by rzikm** [comment]
I think we can simply try opening and closing the API registration, and then lazily reopen it when we actually intend to use it.

**Comment by rzikm** [comment]
I talked with Nick in MsQuic discord about this, `MsQuicOpenVersion` will allocate threads for handling UDP sockets, this is being made lazy by https://github.com/microsoft/msquic/pull/3034, and another set of threads is allocated by `RegistrationOpen`.

One direction they consider moving is allowing applications (in our case, .NET Runtime) completely control the threads/scheduling, but that is distant future. 

Closing MsQuic via `MsQuicClose` after checking for support and later opening it again lazily when needed is currently the best option.

**Comment by wfurt** [comment]
Is it worth of picking up that change? It seems like we can suppress the ongoing thread cost but we would still create and destroy the threads when checking the version...

**Comment by rzikm** [comment]
Reopening for backport

**Comment by karelz** [comment]
The change in 8.0 was reverted by PRT #74984

**Comment by rzikm** [comment]
This will need a change in MsQuic: https://github.com/microsoft/msquic/pull/3041, so backporting to 7.0 would require porting updated MsQuic as well, do we still want to do that? @karelz?

cc: @stephentoub

**Comment by karelz** [comment]
Triage: We plan to get the msquic change into 2.1.1 release.
@rzikm do you think we should start depending on the 2.1.1 version (on Linux), or is that not necessary? (aka users will see failures if they choose not to update msquic)

**Comment by rzikm** [comment]
I think we can keep 2.1.0, it is, after all, the minimum required version.

But I would expect that most people regularly update packages on their systems so most people would run with 2.1.1 anyway.

**Comment by karelz** [comment]
Reopening to track 7.0 backport.
@wfurt are we ready to start the backport, or are we blocked on other changes to msquic 2.1.1 deployment to our Docker images and into Helix?

**Comment by rzikm** [comment]
From what I understand, the PR updates the relevant docker images to a version which works, and later, there will be another PR which updates to images to the official 2.1.1 bits on all tested platforms.

**Comment by karelz** [comment]
Based on discussion on the PR: https://github.com/dotnet/runtime/pull/75330#issuecomment-1243910432 ... we believe some fix should happen in 7.0 - either in RC2/GA or in servicing.

**Comment by karelz** [comment]
Open questions:
- Is #75493 regression caused by unloading msquic and will this change address it?
- CR feedback - take another look if we can avoid touching msquic more - https://github.com/dotnet/runtime/pull/75330#issuecomment-1243921873

**Comment by karelz** [comment]
Fixed in main (8.0) in PRs #75163 and #75441 and in 7.0 (GA) in PR #75521

**Comment by karelz** [comment]
We have customers impacted on 6.0 as well - only via Kestrel. Reopening for 6.0.x backport.

**Comment by karelz** [comment]
Fixed also in 6.0.15 in PR #80785 -- see https://github.com/dotnet/runtime/issues/74629#issuecomment-1256420222 for details.
